lib/liquid/tags/for.rb

```diff
@@ -97,14 +98,15 @@ def render(context)
       length = segment.length
 
       # Store our progress through the collection for the continue flag
-      context.registers[:for][@name] = from + segment.length
+      for_offsets[@name] = from + segment.length
 
-      parent_loop = context['forloop'.freeze]
```

Comment:
Yep, essentially. There might be other weird things you could do with `before_method` as well.